src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -457,6 +464,125 @@ public RealJenkinsRule prepareHomeLazily(boolean prepareHomeLazily) {
+                "      <configuration>\n" +
+                "        <artifactItems>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bc-fips</artifactId>\n" +
+                "            <version>1.0.2.4</version>\n" +
+                "          </artifactItem>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bcpkix-fips</artifactId>\n" +
+                "            <version>1.0.7</version>\n" +
+                "          </artifactItem>\n" +
+                "          <artifactItem>\n" +
+                "            <groupId>org.bouncycastle</groupId>\n" +
+                "            <artifactId>bctls-fips</artifactId>\n" +
+                "            <version>1.0.17</version>\n" +
+                "          </artifactItem>\n" +
+                "        </artifactItems>\n" +
+                "        <stripVersion>true</stripVersion>\n" +
+                "      </configuration>\n" +
+                "    </execution>\n" +
+                "  </executions>\n" +
+                "</plugin>";
+        return new IllegalArgumentException(message);
+    }
+
+    private Path writeFIPSJavaSecurityFile() throws IOException {
+        String javaHome = System.getProperty("java.home");
+        if(javaHome == null) {
+            throw new IllegalArgumentException("Cannot find java.home property");
+        }
+        Path javaSecurity = Paths.get(javaHome, "conf", "security", "java.security");
+        Properties properties = new Properties();
+        Path securityFile = Files.createTempFile("java", ".security");
+        try (InputStream inputStream = Files.newInputStream(javaSecurity);
+             OutputStream outputStream = Files.newOutputStream(securityFile)) {
+            properties.load(inputStream);
+            properties.put("security.provider.1", "org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider C:HYBRID;ENABLE{All};");
+            properties.put("security.provider.2", "org.bouncycastle.jsse.provider.BouncyCastleJsseProvider fips:BCFIPS");
+            properties.put("security.provider.3", "sun.security.provider.Sun");
+            properties.put("fips.provider.1", "org.bouncycastle.jcajce.provider.BouncyCastleFipsProvider C:HYBRID;ENABLE{All};");
+            properties.put("fips.provider.2", "org.bouncycastle.jsse.provider.BouncyCastleJsseProvider fips:BCFIPS");
+            properties.put("keystore.type", "BCFKS");
+            //properties.put("securerandom.strongAlgorithms", "PKCS11:SunPKCS11-NSS-FIPS");
+            properties.put("security.useSystemPropertiesFile", "false");
+            properties.put("ssl.KeyManagerFactory.algorithm", "PKIX");
+            properties.put("fips.keystore.type", "BCFKS");
+            properties.store(outputStream, "");
```

Comment:
this is missing a lot more, you would need to remove all security providers except the 3 (and same for fips..) to get something that does not have the non FIPS providers.

Additionally, the changes here may also be JRE version (and vendor) specific (we needed to make some changes between java 17 and 21 IIRC)

Reply:
removal done.
I have tested this with both 17 and 21 and btw this is building on ci.jenkins.io with 17 and 21.
The idea is to load the current java.security file, modify it to our needs, and then use this modified one only. 
Maybe this will have some issues on some exotic JVM distro manually tailored, but the goal is to have it working for most Jenkins developers and most CIs using this. 
And it's probably the case with this basic, simple approach. 
Let's do this first and let's see if we have issues reported.